src/server/middleware/twitter.middleware.ts

```diff
@@ -304,6 +313,172 @@ function loadTwitterData(
+    in_reply_to_status_id: tweet.entities?.referenced_tweets?.find(
+      (tweet) => tweet.type === 'replied_to'
+    )?.id,
+    lang: tweet.lang,
+    reply_count: tweet.public_metrics.reply_count,
+    retweet_count: tweet.public_metrics.retweet_count,
+    source: tweet.source,
+    truncated: tweet.text.length > 140,
+    user: getUser(tweet.author_id, includes),
+  };
+
+  if (tweetObject.truncated) {
+    // Enteprise populates the extended_tweet field if the tweet is truncated,
+    // so we add it manually here for consistency.
+    tweetObject.extended_tweet = {
+      full_text: tweetObject.text,
+      display_text_range: [0, 0], // Indices not available in v2.
+      entities,
+    };
+  }
+
+  return tweetObject;
+}
+
+function packEntities(
+  tweet: V2TweetObject,
+  includes: V2Includes
+): TweetEntities {
+  const entities: TweetEntities = {};
+
+  if (tweet.entities && tweet.entities.hashtags) {
+    entities.hashtags = tweet.entities.hashtags.map(
+      (hashtag) =>
+        <TweetHashtag>{
+          indices: [hashtag.start, hashtag.end],
+          text: hashtag.tag,
+        }
+    );
+  }
+
+  if (tweet.entities && tweet.entities.urls) {
+    entities.urls = tweet.entities.urls.map(
+      (url) =>
+        <TweetUrl>{
+          display_url: url.display_url,
+          expanded_url: url.extended_url,
+          indices: [url.start, url.end],
+        }
+    );
+  }
+
+  if (tweet.entities && tweet.entities.mentions) {
+    entities.user_mentions = tweet.entities.mentions.map(
+      (mention) =>
+        <TweetUserMention>{
+          id_str: mention.id,
+          indices: [mention.start, mention.end],
+          screen_name: mention.username,
+        }
+    );
+  }
+
+  if (tweet.attachments && tweet.attachments.media_keys) {
+    entities.media = tweet.attachments.media_keys
+      .flatMap((media_key) => {
+        // v2 includes the media fields (like media_url) in a separate
+        // `includes` object, so we search there for the media item in question.
+        const media = includes.media?.find(
+          (media) => media.media_key === media_key
+        );
+        // This shouldn't happen, but we throw an error if it does.
+        if (!media) {
+          throw new Error('Unable to find media');
+        }
+        return <TweetMedia>{
+          indices: [0, 0], // Indices not available in v2.
+          media_url: media.url,
+          type: media.type,
+        };
+      })
+      // Filter for photos because that's all we display.
+      .filter((media) => media.type === 'photo');
+  }
+
+  return entities;
+}
+
+function getUser(id: string, includes: V2Includes): TwitterUser {
+  // v2 includes the user fields (like profile_image_url) in a separate
+  // `includes` object, so we search there for the media item in question.
+  const user = includes.users?.find((user) => user.id === id);
+  if (!user) {
+    throw new Error('Unable to find user');
+  }
+  return {
+    id_str: user?.id,
```

Comment:
Does the compiler complain here if you don't use the '?' ? I would think it might be okay to omit it because of the check directly above this is effectively type narrowing, but I'm not 100% sure.